From 627f4073157fec9abc42334fcc3c96ca3da7d11a Mon Sep 17 00:00:00 2001
From: Venkataprasad Karna <Venkataprasad_Karna@comcast.com>
Date: Thu, 7 Jan 2021 06:31:21 +0000
Subject: [PATCH] setting proper callsign

---
 Source/websocket/JSONRPCLink.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/websocket/JSONRPCLink.h b/Source/websocket/JSONRPCLink.h
index 3eb8b079..e057e930 100644
--- a/Source/websocket/JSONRPCLink.h
+++ b/Source/websocket/JSONRPCLink.h
@@ -594,7 +594,7 @@ namespace JSONRPC {
                 _localSpace = localCallsign;
             }
 
-            uint8_t version = Core::JSONRPC::Message::Version(_callsign);
+            uint8_t version = Core::JSONRPC::Message::Version(callsign + '.');
             if( version != static_cast<uint8_t>(~0) ) {
                 _versionstring = '.' + Core::NumberType<uint8_t>(version).Text();
             }
-- 
2.24.0

