commit 1f954ab65182b7c10abe00bfa15aa027577d94aa (HEAD)
Author: preeja <preeja_raveendran@comcast.com>
Date:   Wed Nov 5 17:45:43 2025 +0530

    Firmware Version interface patch

diff --git a/interfaces/IFirmwareVersion.h b/interfaces/IFirmwareVersion.h
new file mode 100644
index 0000000..8eb4828
--- /dev/null
+++ b/interfaces/IFirmwareVersion.h
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "Module.h"
+
+namespace WPEFramework {
+namespace Exchange {
+
+    struct EXTERNAL IFirmwareVersion : virtual public Core::IUnknown {
+        enum { ID = ID_FIRMWARE_VERSION };
+
+        virtual ~IFirmwareVersion() override = default;
+
+        virtual uint32_t Imagename(string& imagename /* @out */) const = 0;
+        virtual uint32_t Sdk(string& sdk /* @out */) const = 0;
+        virtual uint32_t Mediarite(string& mediarite /* @out */) const = 0;
+        virtual uint32_t Yocto(string& yocto /* @out */) const = 0;
+        virtual uint32_t Pdri(string& pdri /* @out */) const = 0;
+    };
+}
+}
+
diff --git a/interfaces/Ids.h b/interfaces/Ids.h
index a37db24..2932adf 100644
--- a/interfaces/Ids.h
+++ b/interfaces/Ids.h
@@ -264,6 +264,7 @@ namespace Exchange {
         ID_DEVICE_CAPABILITIES_VIDEO                 = ID_DEVICE_INFO + 6,
         ID_DEVICE_CAPABILITIES_VIDEO_OUTPUT          = ID_DEVICE_INFO + 7,
         ID_DEVICE_CAPABILITIES_RESOLUTION            = ID_DEVICE_INFO + 8,
+        ID_FIRMWARE_VERSION                          = ID_DEVICE_INFO + 9,
 
         ID_DIALSERVER                                = RPC::IDS::ID_EXTERNAL_INTERFACE_OFFSET + 0x380,
         ID_DIALSERVER_APPLICATION                    = ID_DIALSERVER + 1,
diff --git a/jsonrpc/DeviceInfo.json b/jsonrpc/DeviceInfo.json
index c0519fc..07cbb14 100644
--- a/jsonrpc/DeviceInfo.json
+++ b/jsonrpc/DeviceInfo.json
@@ -818,6 +818,10 @@
           "yocto": {
             "type": "string",
             "$ref": "#/definitions/yocto"
+          },
+          "pdri": {
+            "type": "string",
+            "example": "PX051AEI_VBN_2203_sprint_20220331225312sdy_NG"
           }
         },
         "required": [
