commit 34aade59f285b69802e2568d13be5d04cbc4bf49 (HEAD)
Author: preeja <preeja_raveendran@comcast.com>
Date:   Wed Nov 5 16:31:00 2025 +0530

     R4.4.1 L2 test AddPdriApi.patch

diff --git a/interfaces/IFirmwareVersion.h b/interfaces/IFirmwareVersion.h
index 0a3439f..8eb4828 100644
--- a/interfaces/IFirmwareVersion.h
+++ b/interfaces/IFirmwareVersion.h
@@ -14,6 +14,7 @@ namespace Exchange {
         virtual uint32_t Sdk(string& sdk /* @out */) const = 0;
         virtual uint32_t Mediarite(string& mediarite /* @out */) const = 0;
         virtual uint32_t Yocto(string& yocto /* @out */) const = 0;
+        virtual uint32_t Pdri(string& pdri /* @out */) const = 0;
     };
 }
 }
diff --git a/jsonrpc/DeviceInfo.json b/jsonrpc/DeviceInfo.json
index c0519fc..07cbb14 100644
--- a/jsonrpc/DeviceInfo.json
+++ b/jsonrpc/DeviceInfo.json
@@ -818,6 +818,10 @@
           "yocto": {
             "type": "string",
             "$ref": "#/definitions/yocto"
+          },
+          "pdri": {
+            "type": "string",
+            "example": "PX051AEI_VBN_2203_sprint_20220331225312sdy_NG"
           }
         },
         "required": [
