 interfaces/IDeviceInfo2.h |  3 ++-
 jsonrpc/DeviceInfo.json   | 22 ++++++++++++++++++++++
 2 files changed, 24 insertions(+), 1 deletion(-)
diff --git a/interfaces/IDeviceInfo2.h b/interfaces/IDeviceInfo2.h
index f87914d..90e9c41 100644
--- a/interfaces/IDeviceInfo2.h
+++ b/interfaces/IDeviceInfo2.h
@@ -18,6 +18,7 @@ namespace Exchange {
         virtual uint32_t Model(string& model /* @out */) const = 0;
         virtual uint32_t DeviceType(string& deviceType /* @out */) const = 0;
         virtual uint32_t DistributorId(string& distributorId /* @out */) const = 0;
+        virtual uint32_t Brand(string& brand /* @out */) const = 0;
     };
 
     struct EXTERNAL IDeviceAudioCapabilities : virtual public Core::IUnknown {
@@ -71,4 +72,4 @@ namespace Exchange {
         virtual uint32_t SupportedHdcp(const string& videoDisplay /* @in */, CopyProtection& supportedHDCPVersion /* @out */) const = 0;
     };
 }
-}
\ No newline at end of file
+}
diff --git a/jsonrpc/DeviceInfo.json b/jsonrpc/DeviceInfo.json
index ed32927..d927008 100644
--- a/jsonrpc/DeviceInfo.json
+++ b/jsonrpc/DeviceInfo.json
@@ -206,6 +206,28 @@
       "description": "Yocto version",
       "example": "dunfell"
     },
+    "brandname": {
+      "summary": "Device brand",
+      "readonly": true,
+      "params": {
+        "type": "object",
+        "properties": {
+          "brand": {
+            "type": "string",
+            "example": "Comcast"
+          }
+        },
+        "required": [
+          "brand"
+        ]
+      },
+      "errors": [
+        {
+          "description": "General error",
+          "$ref": "#/common/errors/general"
+        }
+      ]
+    },
     "distributorid": {
       "type": "string",
       "enum": [
