From: preeja <preeja_raveendran@comcast.com>
Date: Wed, 5 Nov 2025 21:12:19 +0530
Subject: [PATCH] RDK-59565: Add new property named "pdri"

Signed-off-by: preeja <preeja_raveendran@comcast.com>
---
 interfaces/IFirmwareVersion.h | 1 +
 jsonrpc/DeviceInfo.json       | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/interfaces/IFirmwareVersion.h b/interfaces/IFirmwareVersion.h
index ba6ce5a..db792e1 100644
--- a/interfaces/IFirmwareVersion.h
+++ b/interfaces/IFirmwareVersion.h
@@ -14,6 +14,7 @@ namespace Exchange {
         virtual uint32_t Sdk(string& sdk /* @out */) const = 0;
         virtual uint32_t Mediarite(string& mediarite /* @out */) const = 0;
         virtual uint32_t Yocto(string& yocto /* @out */) const = 0;
+        virtual uint32_t Pdri(string& pdri /* @out */) const = 0;
     };
 }
 }
diff --git a/jsonrpc/DeviceInfo.json b/jsonrpc/DeviceInfo.json
index ed32927..6173e74 100644
--- a/jsonrpc/DeviceInfo.json
+++ b/jsonrpc/DeviceInfo.json
@@ -468,6 +468,10 @@
           "yocto": {
             "type": "string",
             "$ref": "#/definitions/yocto"
+          },
+          "pdri": {
+            "type": "string",
+            "example": "COESST11AEI_PDRI_PROD_20240119170804_3.2.2.0.bin"
           }
         },
         "required": [
-- 
2.34.1
