diff --git a/interfaces/Ids.h b/interfaces/Ids.h
index a37db24..2932adf 100644
--- a/interfaces/Ids.h
+++ b/interfaces/Ids.h
@@ -264,6 +264,7 @@ namespace Exchange {
         ID_DEVICE_CAPABILITIES_VIDEO                 = ID_DEVICE_INFO + 6,
         ID_DEVICE_CAPABILITIES_VIDEO_OUTPUT          = ID_DEVICE_INFO + 7,
         ID_DEVICE_CAPABILITIES_RESOLUTION            = ID_DEVICE_INFO + 8,
+        ID_FIRMWARE_VERSION                          = ID_DEVICE_INFO + 9,

         ID_DIALSERVER                                = RPC::IDS::ID_EXTERNAL_INTERFACE_OFFSET + 0x380,
         ID_DIALSERVER_APPLICATION                    = ID_DIALSERVER + 1,

diff --git a/interfaces/IFirmwareVersion.h b/interfaces/IFirmwareVersion.h
new file mode 100644
index 0000000..ba6ce5a
--- /dev/null
+++ b/interfaces/IFirmwareVersion.h
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "Module.h"
+
+namespace WPEFramework {
+namespace Exchange {
+
+    struct EXTERNAL IFirmwareVersion : virtual public Core::IUnknown {
+        enum { ID = ID_FIRMWARE_VERSION };
+
+        virtual ~IFirmwareVersion() override = default;
+
+        virtual uint32_t Imagename(string& imagename /* @out */) const = 0;
+        virtual uint32_t Sdk(string& sdk /* @out */) const = 0;
+        virtual uint32_t Mediarite(string& mediarite /* @out */) const = 0;
+        virtual uint32_t Yocto(string& yocto /* @out */) const = 0;
+    };
+}
+}
