From 93d3ece5e158203ca80e9511d21f149ab61fea78 Mon Sep 17 00:00:00 2001
From: praveenkumar_senthilkumar <PraveenKumar_SenthilKumar@comcast.com>
Date: Wed, 8 Feb 2023 10:22:42 +0000
Subject: [PATCH] Initialization-fixed

---
 HdmiCec/HdmiCec.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/HdmiCec/HdmiCec.h b/HdmiCec/HdmiCec.h
index a19a5114..66e930a1 100644
--- a/HdmiCec/HdmiCec.h
+++ b/HdmiCec/HdmiCec.h
@@ -59,8 +59,9 @@ namespace WPEFramework {
                bool m_isOSDNameUpdated;
                bool m_isVendorIDUpdated;

+               /*CID-283052-Initializing m_deviceInfoStatus*/
                CECDeviceInfo()
-               : m_logicalAddress(0),m_vendorID(0,0,0),m_osdName("NA"), m_isOSDNameUpdated (false), m_isVendorIDUpdated (false)
+               : m_logicalAddress(0),m_vendorID(0,0,0),m_osdName("NA"), m_deviceInfoStatus(0), m_isOSDNameUpdated (false), m_isVendorIDUpdated (false)
                {
                        BITMASK_CLEAR(m_deviceInfoStatus, 0xFFFF); //Clear all bits
                }
--
2.17.1
